src/ucp/api/ucp.h

```diff
@@ -4636,6 +4637,11 @@ typedef struct ucp_ep_attr {
      * UCS_ERR_NOT_CONNECTED will be returned.
      */
     struct sockaddr_storage remote_sockaddr;
+
+    /**
+     * Pointer to struct containing transport and device names used by this endpoint.
+     */
+    ucp_transports_t *transports;
```

Comment:
```
struct {
   const char *transport_name;
   const char *device_name;
   
} ucp_transport_entry_t;

struct ucp_ep_attr {
  ...
  ucp_transports_t transports;
};

struct ucp_transports {
   ucp_transport_entry_t *entries;
   unsigned     num_entries;
   // ucp should not fill more than this; and use it to access the array instead of by index
   size_t        entry_size; 
};

----
user code:
attr.transports.entry_size=sizeof(ucp_transport_entry_t);
attr.transports.num_entries = COUNT;
attr.transports.entries = alloca(COUNT*sizeof(*attr.transports.entries ));



```